### PR TITLE
`impl Display` for pdg

### PR DESCRIPTION
**Pull request by kkysen**
This refactors `c2rust-pdg`'s `main.rs` code so that the printing logic in `main` is moved to an `impl Display for PdgRepr`.  Thus, we can either `println!` it for `main` or `.to_string()` it for `insta::assert_display_snapshot!` (not yet implemented in this PR; see #636).

### CODE REVIEW

File: pdg/src/main.rs
Comment (aneksteind) [inline pdg/src/main.rs:72]: I think `PdgRepr` should be constructed from a `Pdg`, not the other way around

Comment (aneksteind) [reply pdg/src/main.rs:72]: i.e. `PdgRepr::new` should do this IMO

Comment (kkysen) [reply pdg/src/main.rs:72]: Can `PdgRepr::new` be the real implementation and then still have a `Pdg::repr` method?

File: pdg/src/main.rs
Comment (fw-immunant) [inline pdg/src/main.rs:130]: I would move this `if` to surround the entire `for graph...` loop, and then make this `writeln!` unconditional. That would clarify that the `writeln!(f)` terminates each graph printed and would avoid iterating if we don't need to look at graphs.

File: pdg/src/main.rs
Comment (fw-immunant) [inline pdg/src/main.rs:121]: This can be moved inside of `if should_print(ToPrint::WritePermissions) { ... }`.

**Review by fw-immunant — APPROVED** [review]
A couple refactors suggested, but not blocking merge. LGTM overall.

File: pdg/src/main.rs
Comment (kkysen) [reply pdg/src/main.rs:72]: Actually, the fields of `PdgRepr` are `pub`, so it can be constructed on its own.  A `PdgRepr::new` wouldn't really offer any advantage, and I'd like to keep `Pdg::repr` for ergonomics.

File: pdg/src/main.rs
Comment (kkysen) [reply pdg/src/main.rs:130]: I'd like to change this in a separate PR, since this right now is just moving code, not changing anything.

File: pdg/src/main.rs
Comment (kkysen) [reply pdg/src/main.rs:121]: (Same with this) I'd like to change this in a separate PR, since this right now is just moving code, not changing anything.

File: pdg/src/main.rs
Comment (kkysen) [reply pdg/src/main.rs:130]: Wait, this is not quite equivalent.  If `--print write-permissions` is specified, but no `--print graphs`, the write permissions and no graphs should be printed.

Comment (fw-immunant) [reply pdg/src/main.rs:130]: A separate PR for these changes is fine.

I think what I requested is equivalent, though--the three `writeln!`s in the loop are guarded by `should_print(ToPrint::Graphs)`, `should_print(ToPrint::WritePermissions)`, and their disjunction, respectively. Making the whole loop conditional on the disjunction doesn't impact the first two conditionals as they're refinements of it, but it makes the third fully redundant, so it can be made unconditonal:
```rust
        if should_print(ToPrint::Graphs) || should_print(ToPrint::WritePermissions) {
            for graph in &graphs.graphs {
                let needs_write = graph
                    .needs_write_permission()
                    .map(|node_id| node_id.as_usize())
                    .collect::<Vec<_>>();
                if should_print(ToPrint::Graphs) {
                    writeln!(f, "{graph}")?;
                }
                if should_print(ToPrint::WritePermissions) {
                    writeln!(f, "nodes_that_need_write = {needs_write:?}")?;
                }
                writeln!(f)?;
            }
        }
```

Comment (kkysen) [reply pdg/src/main.rs:130]: Oh like that you mean.  I'm not sure that's much of an advantage, since it's just hosting the `if` around the `writeln!(f)` to outside the loop.

Comment (fw-immunant) [reply pdg/src/main.rs:130]: The concrete advantage is that it skips iterating over the graphs if nothing about them needs to be printed. It's not a huge deal, but is a small "pay for what you use".

File: pdg/src/main.rs
Comment (aneksteind) [reply pdg/src/main.rs:72]: I'm not sold on the utility of `Pdg` knowing anything about what a `PdgRepr` is, that's my main concern. I think it should be constructed on its own, how is less important.

Comment (kkysen) [reply pdg/src/main.rs:72]: I see `PdgRepr` more as an `impl Display` for `Pdg`, but with `&[ToPrint]`s, too.  I'm not sure why `Pdg` can't know about `PdgRepr`, especially as its such a simple and transparent struct.

Comment (kkysen) [reply pdg/src/main.rs:73]: I'm going to merge this as is if that's okay.  We can easily change it back afterward if we need to as it's only a couple of lines of code.

File: pdg/src/main.rs
Comment (kkysen) [reply pdg/src/main.rs:131]: Fixed in #645.

**Pull request merged**
